### DISCUSSION TITLE
Ynoa wireless knop dimmer

### DISCUSSION THREAD

**Discussion by ekesgin** [Koenkk/zigbee2mqtt#10347] [Devices]
hi,

I can't add the device ZBT-DIMSwitch-D0000. This is a wireless knob dimmer.  I've bought id from this webshop. 
https://www.ledlampenkopen.nu/ynoa-1-knops-afstandsbediening-zigbee-3-0.html

It is not supported in Z2M yet.  I've followed the instruction from de site https://www.zigbee2mqtt.io/advanced/support-new-devices/01_support_new_devices.html#_2-adding-your-device. But I can't add the device. 

I 'm running the official docker add-on under Home Assistant.  This what I'm getting on add-on log. 

```Received Zigbee message from 'ZBT-DIMSwitch-D0000', type 'commandOff', cluster 'genOnOff', data '{}' from endpoint 1 with groupID 16394
Warning Received message from unsupported device with Zigbee model 'ZBT-DIMSwitch-D0000' and manufacturer name 'LDS'
Warning Please see: https://www.zigbee2mqtt.io/advanced/support-new-devices/01_support_new_devices.html`

and this is my javascript file according the instruction. 

`const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const e = exposes.presets;
const ea = exposes.access;

const definition = {
    zigbeeModel: ['LDS'], // The model ID from: Device with modelID 'lumi.sens' is not supported.
    model: 'ZBT-DIMSwitch-D0000', // Vendor model number, look on the device for a model number
    vendor: 'Ynoa', // Vendor of the device (only used for documentation and startup logging)
    description: 'Ynoa Remote control One Button Dimmer', // Description of the device, copy from vendor site. (only used for documentation and startup logging)
    fromZigbee: [fz.genLevelCtrl'], // We will add this later
    toZigbee: [], // Should be empty, unless device can be controlled (e.g. lights, switches).
    exposes: [e.battery(), e.linkquality(), e.genLevelCtrl()], // Defines what this device exposes, used for e.g. Home Assistant discovery and in the frontend
};

module.exports = definition;`

And this is my configuration.yaml file.
```

**Comment by Koenkk** [comment]
Replace `zigbeeModel: ['LDS'],` with `zigbeeModel: ['ZBT-DIMSwitch-D0000'],`, with `fromZigbee: [fz.command_on, fz.command_off, fz.command_move, fz.command_stop, fz.battery],` I expect that you should get actions from this device.

**Comment by ekesgin** [comment]
Hi Koenkk,

Thanks for your advice. I'm getting now some actions from the device. This is some output from z2m log.

![image](https://user-images.githubusercontent.com/25054985/147386410-aff98780-d710-472b-a6e8-e5a6f6ff4cfc.png)

I see couple of attributes. but the battery value is not shown.
![image](https://user-images.githubusercontent.com/25054985/147386519-60f09282-60e7-4413-b936-7731cab29fbb.png)

If I want create an automation bay I see one `entity_id: sensor.zbt_dimswitch_d0000_battery.` Actually I'm expecting more entities from device like "action".  Finally I want to dim some of my lights with this device.

This what I' m on HA frontend seeing.

![image](https://user-images.githubusercontent.com/25054985/147387076-d676bbcf-946b-4c1c-81af-00bdcd1b41b7.png)

I think I'm forgetting something bu what?

happy holidays btw.

**Comment by Koenkk** [comment]
```js
const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const e = exposes.presets;
const ea = exposes.access;

const definition = {
    zigbeeModel: ['ZBT-DIMSwitch-D0000'],
    model: '8718801528334',
    vendor: 'Ynoa',
    description: 'Remote control one button dimmer',
    fromZigbee: [fz.command_on, fz.command_off, fz.command_move, fz.command_stop, fz.battery],
    exposes: [e.action(['on', 'off', 'brightness_move_up', 'brightness_move_down', 'brightness_stop']), e.battery()],
    toZigbee: [],
    configure: async (device, coordinatorEndpoint, logger) => {
        const endpoint = device.getEndpoint(1);
        await reporting.bind(endpoint, coordinatorEndpoint, ['genPowerCfg']);
        await reporting.batteryPercentageRemaining(endpoint);
    },
};

module.exports = definition;
```

Should do it, battery may take up to 1 day to report

**Comment by ekesgin** [comment]
Hi Koenkk, It's working now. Thank you for helping.  Next step is building an home assistant automation. Of course the automations belongs to HA forum.

**Comment by Koenkk** [comment]
Added this device.

Changes will be available in the dev branch in a few hours from now. (https://www.zigbee2mqtt.io/advanced/more/switch-to-dev-branch.html)

**Comment by bartplessers** [comment]
Hi @ekesgin,

Have you ever tried the IKEA Symfonisk remote?

https://www.ikea.com/us/en/p/symfonisk-sound-remote-white-20370482/

You can configure it as dimmer, but I'm not really happy with the hardware: hard to control a light and not "responsive": you can assign a "speed" when turning the knob, but it never feels like a real dimmer.
I would also like to have some more "physical resistance" on the rotary dimmer.


What are your findings on your device ?

I'm afraid that you can't simulate a "fast" turn on the knob as it exposes only
- brightness_move_down
- brightness_move_up

but no "angle speed"

I would love to find a rotary knob that also has those properties. So, let's say, if you turn it only 5 degrees, it will send "brightness increase: 10" and if you turn it quickly 90 degrees, it would send "brightness increase: 180" or something like that. Any idea of such a device?

I'm looking for a while now to find some devices that can replace my old dimmers, but no success yet...

Kind regards,
Bart

**Reply by ekesgin** [reply]
hi bartplessers,  
I didnt tried ikea IKEA Symfonisk remote?

But I am not satisfied with functionality of the Ynoa dimmer. I think it is similar as ikea symphonisk. The only difference is that it has also an on/off switch.
As you noticed there is no resistance when turning left and right.

But today I received another dimmer wich I’ve ordered on Aliexpress a few weeks ago. It took a month to get it today. This dimmer has a little bit resistance if you turn it left or right. You can feel the gear. It gives a beter feelings when you turning it. But I haven't gotten it to work yet. Zigbee2mqtt recognizes it as an "unsupported device" with the model TS004F and the manufacturer name __TZ3000_. I will try to get working with zigbee2mqtt in the next days..
You can find more information about this dimmer through this sites.  .

https://hindi-alibaba-com.translate.goog/p-detail/tuya-zigbee-3-0-wireless-remote-control-module-smart-bedroom-wifi-electronic-intelligent-home-automation-wireless- scene-switch-1600397501884.html?_x_tr_sl=auto&_x_tr_tl=en&_x_tr_hl=en

https://expo.tuya.com/product/935005

I’ve orderd it on aliexpress. 

kind regerds

**Reply by bartplessers** [reply]
Thanx for feedback!

I'm looking forward to your findings on that other knob!

Grtz
B


Ps
To be correct: the IKEA device has also on/off.

**Comment by yannok** [comment]
Hello,
I have the same device from the same seller : 
ZBT-DIMSwitch-D0000 by LDS 
but I only have access to the following values:
Battery, Identify, LQI

I can't find on, off, brightness : 

![image](https://github.com/Koenkk/zigbee2mqtt/assets/5036888/274259a7-e8ec-417d-b1a3-d3a4b63a583f)

How can I troubleshoot this ? Thanks!

**Comment by mrmaximas** [comment]
@Koenkk hi! Can you please help to determine if this device supports direct binding?
I tried it, the on/off and brightness direct commands like from IKEA RODRET don't work ;(
```
{
  "id": 140,
  "type": "EndDevice",
  "ieeeAddr": "0x588e81fffe2111d5",
  "nwkAddr": 64209,
  "manufId": 4456,
  "manufName": "LDS",
  "powerSource": "Battery",
  "modelId": "ZBT-DIMSwitch-D0000",
  "epList": [
    1
  ],
  "endpoints": {
    "1": {
      "profId": 260,
      "epId": 1,
      "devId": 2080,
      "inClusterList": [
        0,
        1,
        3,
        2821,
        4096,
        64769
      ],
      "outClusterList": [
        3,
        4,
        6,
        8,
        25,
        4096
      ],
      "clusters": {
        "genBasic": {
          "attributes": {
            "stackVersion": 0,
            "hwVersion": 2,
            "dateCode": "20200619-47",
            "swBuildId": "2.22"
          }
        },
        "genPowerCfg": {
          "attributes": {
            "batteryPercentageRemaining": 200
          }
        }
      },
      "binds": [],
      "configuredReportings": [
        {
          "cluster": 1,
          "attrId": 33,
          "minRepIntval": 3600,
          "maxRepIntval": 65000,
          "repChange": 0
        }
      ],
      "meta": {}
    }
  },
  "appVersion": 38,
  "stackVersion": 0,
  "hwVersion": 2,
  "dateCode": "20200619-47",
  "swBuildId": "2.22",
  "zclVersion": 3,
  "interviewCompleted": true,
  "interviewState": "SUCCESSFUL",
  "meta": {
    "configured": -1377615576
  },
  "lastSeen": 1745685077257
}
```

**Comment by edwin-2023** [comment]
Here a Ynoa One Button Dim item LA-AAN-UIT-7A.
In zigbee2mqtt at device/toont is see on/off when short press. Long press gives brightness_move_down, brightness_move_stop, brightness_move_up but in homw assistent mosquito broker only has battery and linkquality.
How to show the actions in HA so i can use the button?

**Reply by edwin-2023** [reply]
Really, no one?

**Comment by edwin-2023** [comment]
So, please help me out on this! I've got a few of this button's and it would be a shame to throw them into the bin...

**Reply by mrmaximas** [reply]
i have six same shitknobs, it's works with z2m but without direct binding.

**Comment by ekesgin** [comment]
hi, 
i have 2 of these devices. Both work ust fine. 

This is screen shot of mqtt broker in HA 
![image](https://github.com/user-attachments/assets/99ec9604-0d66-4cb5-8f72-063239b9b039)


and this is from zigbee2mqtt. 
![image](https://github.com/user-attachments/assets/2e52f629-255c-4a66-8d2c-0108d381605d)

And this is an automation that works fine.

`alias: test ZBT
description: ""
triggers:
  - domain: mqtt
    device_id: 404a7ba36f45764c91e6891cfbb700e2
    type: action
    subtype: "on"
    trigger: device
  - domain: mqtt
    device_id: 404a7ba36f45764c91e6891cfbb700e2
    type: action
    subtype: "off"
    trigger: device
conditions: []
actions:
  - choose:
      - conditions:
          - condition: state
            entity_id: event.zbt_dimswitch_d0000_action
            attribute: event_type
            state: "on"
        sequence:
          - action: light.turn_on
            metadata: {}
            data: {}
            target:
              entity_id: light.kantoor_dimmer
      - conditions:
          - condition: state
            entity_id: event.zbt_dimswitch_d0000_action
            attribute: event_type
            state: "off"
        sequence:
          - action: light.turn_off
            metadata: {}
            data: {}
            target:
              entity_id: light.kantoor_dimmer
mode: single`

**Comment by edwin-2023** [comment]
I only see this in HA
![image](https://github.com/user-attachments/assets/4f1bdde7-22d8-4c4d-840c-25cf7dad88d5)

and in zigbee2mqtt
![image](https://github.com/user-attachments/assets/156b196d-fcdb-4b3b-ae68-137e3e04da4a)

So how did you get action in diagnostics?

**Comment by edwin-2023** [comment]
Found diagnostics, is translated to "Toont" Argh :(
![image](https://github.com/user-attachments/assets/8648cda6-4618-4718-b4e8-b77d12e1e8d2)
So. zigbee2mqtt has action but somehow the action attribute does not arrive in HomeAssistant.
Really not a clue how to solve this, please help...

**Reply by ekesgin** [reply]
Hi, maybe you accidentally disabled the action entity in the device list. I see on your screenshot "laat meer zien". Just click there to see what other attributes are there.

**Comment by edwin-2023** [comment]
Unfortunaly it's not so simple. the screenshot said minder = less so all entities are visible
![image](https://github.com/user-attachments/assets/afdf4612-525b-4295-874e-db1166ab88a2)
![image](https://github.com/user-attachments/assets/3f50f223-6f50-4fbd-8062-0d4b1889c974)

One funny thing i noticed is that the device is categoried as "Sensor" and perhaps a sensor cannot have a switch. If that's the case how transform from sensor to switch (or something that shows a switch)
![image](https://github.com/user-attachments/assets/6e48c938-fb3e-4ceb-a4e9-c5c8f38aaa95)

**Reply by ekesgin** [reply]
I have no idea why you don't see the action entity. If you haven't tried it yet, you might try removing the device and adding it again. (edit: now in English :-)  )

**Reply by edwin-2023** [reply]
Just tried, no success but i now realize i used the same name as before. Wil try later again with;
- remove
- reboot ha
- include with different name

**Comment by edwin-2023** [comment]
Solution!
So silly of me. Despite the fact you don't see the action as entity you can select is with an automation....
<img width="470" height="596" alt="image" src="https://github.com/user-attachments/assets/441665ae-5153-4728-82af-812dc9ba6be9" />
